packages/controllers/src/snaps/SnapController.test.ts

```diff
@@ -1849,4 +1928,106 @@ describe('SnapController', () => {
       ).rejects.toThrow('invalid Snap version range');
     });
   });
+
+  describe('_fetchSnap', () => {
+    const sourceCode = '// foo';
+    const shasum = 'vCmyHWIgnBwgiTqSXnd7LI7PbXSQim/JOotFfXkjAQk=';
+    it('can fetch NPM snaps', async () => {
+      const controller = getSnapController();
+
+      const result = await controller.add({
+        origin: FAKE_ORIGIN,
+        id: FAKE_SNAP_ID,
+      });
+      expect(result).toStrictEqual(
+        getSnapObject({
+          sourceCode,
+          status: SnapStatus.installing,
+          manifest: getSnapManifest({ shasum }),
+        }),
+      );
+    });
+
+    it('can fetch local snaps', async () => {
+      const controller = getSnapController();
+
+      fetchMock
+        .mockResponseOnce(
+          JSON.stringify(
+            getSnapManifest({
+              shasum,
+            }),
+          ),
+        )
+        .mockResponseOnce(sourceCode);
+
+      const id = 'local:https://localhost:8081';
+      const result = await controller.add({
+        origin: FAKE_ORIGIN,
+        id,
+      });
+      expect(fetchMock).toHaveBeenCalled();
```

Comment:
You can use `toHaveBeenCalledTimes(2)` here

Reply:
done! it's actually called `3` times, but left a comment to explain that.